### ISSUE TITLE
AirloopHVAC:UnitarySystem Unitary System Total Heating/Cooling Rate sometimes zero when coils are operating

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4976]
Helpdesk ticket 10559
From user:
When I have only one "UnitaryHeatPump:AirToAir:MultiSpeed" system in the model, the unitary system total cooling/heating rate output works ok. But when I have two "UnitaryHeatPump:AirToAir:MultiSpeed" systems in the model, the unitary system total cooling/heating rate output shows zero for one unit even though both the "Cooling Coil Total Cooling Rate" and "Air System Total Cooling Energy" report non-zero numbers. From the mean zone air temperature, it seems that the unitary systems work just fine. Only the reporting for the unitary system total cooling/heating rate has some problem.

Checked example file HVACTemplate-5ZoneUnitarySystem, and these output do report values for all systems, but sometimes they are zero when a coil is operating.  So, it's not just due to multiple systems, but something else, perhaps.

**Comment by Nigusse** [comment]
@mjwitte The cause for this defect is initialization of report variables of the parent object only. Instead of initializing the report variables of current multi-speed heat pump only, it initializes the entire heat pump array, hence report variables of simulated heat pumps are also reset to zero.  As the result only the last simulated heat pump only retains results of these report variables. Below is the incorrect code snippet:

```
    for ( auto & e : MSHeatPump ) {
        e.TotHeatEnergyRate = 0.0;
        e.SensHeatEnergyRate = 0.0;
        e.LatHeatEnergyRate = 0.0;
        e.TotCoolEnergyRate = 0.0;
        e.SensCoolEnergyRate = 0.0;
        e.LatCoolEnergyRate = 0.0;
    }
```

As noted in the help desk if there is single multi-speed heat pump in the idf, then this defect won't be seen.

**Comment by Nigusse** [comment]
In Unitary System (example file HVACTemplate-5ZoneUnitarySystem), the problem is caused by a logic that does not capture all conditions. This problem is different from the multi-speed heat pump defect.

**Comment by Nigusse** [comment]
In the CI machine run only one example file "UnitarySystem_WaterCoils_wMultiSpeedFan.idf" showed diffs in some of the following unitary system report variables. Diffs are expected in the following report variables of Unitary Systems and Multi Speed Air To Air Heat Pumps:

Unitary System Total Cooling Rate
Unitary System Sensible Cooling Rate
Unitary System Latent Cooling Rate
Unitary System Total Heating Rate
Unitary System Sensible Heating Rate
Unitary System Latent Heating Rate

Other example files may show diffs if they have more than one Multi Speed Air To Air Heat Pumps, and Unitary Systems where these report variables are included in the IDF.  Will issue a PR for review.

**Comment by Myoldmopar** [comment]
Closed via #5407
